### PR TITLE
Update react-native-share-menu for 5.0 by @PhilBeaudoin

### PR DESCRIPTION
**Pull request by PhilBeaudoin**
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/meedan/react-native-share-menu/blob/master/API_DOCS.md
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by PhilBeaudoin** [comment]
Hi @hmajid2301 ! Was wondering if you could give this a quick review?

File: types/react-native-share-menu/index.d.ts
Comment (hmajid2301) [inline types/react-native-share-menu/index.d.ts:3]: You should add your name underneath no need to remove mine.

**Review by hmajid2301 — CHANGES_REQUESTED** [review]
add that line back in (sorry for late review).

File: types/react-native-share-menu/index.d.ts
Comment (PhilBeaudoin) [reply types/react-native-share-menu/index.d.ts:3]: Done. Thanks for the guidance.

**Review by hmajid2301 — APPROVED** [review]
LGTM!

**Comment by PhilBeaudoin** [comment]
It seems like the test are failing outside my code in an unrelated package. Any idea why that may, why it’s blocking this PR and how I could fix it?

**Comment by PhilBeaudoin** [comment]
@typescript-bot @hmajid2301 
The CI failures are in module `types/carbon-components-react`, not in the module I update in this PR (`types/react-native-share-menu/index.d.ts`). I'm guessing it was a fluke and would like to rerun the checks but I cant figure out a way to do that. What would you suggest?

**Comment by lukebrandonfarrell** [comment]
Any update on this? Seems like CI failure are not related to this module

**Comment by PhilBeaudoin** [comment]
@lukebrandonfarrell Indeed, it seems the CI failure have nothing to do with the module. I don't know how to land it anyway. Any way you can help?

**Comment by lukebrandonfarrell** [comment]
@PhilBeaudoin maybe a rebase will solve this? If not we need to @ one of the maintainers

**Comment by PhilBeaudoin** [comment]
Hi @hmajid2301 : Nothing has changed, this is just a rebase to fix the CI problems.

**Review by hmajid2301 — APPROVED** [review]

**Comment by PhilBeaudoin** [comment]
Ready to merge

**Comment by lukebrandonfarrell** [comment]
Whoop!

**Pull request merged**
